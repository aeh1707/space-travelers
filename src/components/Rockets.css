.rocket {
  display: flex;
  justify-content: space-evenly;
  margin: 20px 0;
}

.rocket-image {
  width: 20vw;
}

.rocket-info {
  width: 70vw;
}
