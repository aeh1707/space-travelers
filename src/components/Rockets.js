import './Rockets.css';
import { useDispatch } from 'react-redux';
import { PropTypes } from 'prop-types';
import Nav from './Nav';
import { reserveRocket, leaveRocket } from '../redux/rockets/rockets';

const Rockets = (props) => {
  const { rockets } = props;
  console.log(rockets);
  const dispatch = useDispatch();
  const handleReserveRocket = (id) => {
    dispatch(reserveRocket(id));
  };
  const handleLeaveRocket = (id) => {
    dispatch(leaveRocket(id));
  };

  return (
    <>
      <Nav />
      <ul id="rockets">
        {rockets.map((rocket) => (
          <li className="rocket" key={rocket.id}>
            <img className="rocket-image" src={rocket.flickr_images} alt={`${rocket.rocket_name} rocket`} />
            <div className="rocket-info">
              <h2 className="rocket-name">{rocket.rocket_name}</h2>
              <p className="rocket-description">
                {rocket.reserved && <span className="reserved-badge">Reserved</span> }
                &nbsp;
                {rocket.description}
              </p>
              {!rocket.reserved && <button type="button" className="rocket-reserve btn" onClick={() => handleReserveRocket(rocket.id)}>Reserver Rocket</button>}
              {rocket.reserved && <button type="button" className="rocket-cancel btn" onClick={() => handleLeaveRocket(rocket.d)}>Cancel Reservation</button>}
            </div>
          </li>
        ))}
      </ul>
    </>
  );
};

Rockets.propTypes = {
  rockets: PropTypes.instanceOf(Array).isRequired,
};

export default Rockets;
