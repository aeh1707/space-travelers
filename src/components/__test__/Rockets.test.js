import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import Rockets from '../Rockets';
import store from '../../redux/configureStore';
import rocketReducer from '../../redux/rockets/rockets';
import fetchRockets from '../../redux/rockets/rocketsFetch';

describe('Jest test', () => {
  it('Match snapshot', () => {
    const { container } = render(
      <MemoryRouter>
        <Provider store={store}>
          <Rockets />
        </Provider>
      </MemoryRouter>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('Output should be empty state', () => {
    const state = rocketReducer(undefined, {});
    expect(state).toEqual([]);
  });

  it('should return content from api', () => {
    const result = fetchRockets();
    expect(result).not.toEqual(null);
  });
});
