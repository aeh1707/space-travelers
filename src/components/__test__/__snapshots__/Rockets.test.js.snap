// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Jest test Match snapshot 1`] = `
<ul
  id="rockets"
/>
`;
